### PR TITLE
done

### PR DESCRIPTION
**Pull request by Richaa2**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,79 @@
...
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String SUPPLY = "supply,";
+    static final String BUY = "buy,";
+    static final String RESULT = "result,";
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:13]:
```suggestion
    private static final String SUPPLY = "supply";
    private static final String BUY = "buy";
    private static final String RESULT = "result";
    private static final String SEPARATOR = ",";
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,79 @@
...
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File fromMyFile = new File(fromFileName);
+        File toMyFile = new File(toFileName);
+        try {
+            FileReader fileReader = new FileReader(fromMyFile);
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:19]: Lets create separate method for read from file

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,79 @@
...
+                    .append(System.lineSeparator())
+                    .append(BUY).append(buySum)
+                    .append(System.lineSeparator()).append(RESULT)
+                    .append(resultSum).toString();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void createFile(BufferedWriter bufferedWriter, String report) {
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:70]:
```suggestion
    private void writeToFile(BufferedWriter bufferedWriter, String report) {
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,79 @@
...
+        File fromMyFile = new File(fromFileName);
+        File toMyFile = new File(toFileName);
+        try {
+            FileReader fileReader = new FileReader(fromMyFile);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String report = createReport(bufferedReader);
+
+            FileWriter fileWriter = new FileWriter(toMyFile);
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+            createFile(bufferedWriter, report);
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:25]:
```suggestion
            writeToFile(report);
```
lets create fileWriter inside writeToFile method

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,79 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String createReport(BufferedReader reader) {
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:31]:
```suggestion
    private String createReport(List<String> readFromFile) {
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,79 @@
...
+
+            String[] filter = stringBuilder.toString().split(System.lineSeparator());
+            int indexSupply = 7;
+            int indeBuy = 4;
+            for (String result : filter) {
+                if (result.startsWith(SUPPLY)) {
+                    supplySum += Integer.parseInt(result.substring(indexSupply));
+                }
+                if (result.startsWith(BUY)) {
+                    buySum += Integer.parseInt(result.substring(indeBuy));
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:53]:
```suggestion
            for (String result : filter) {
                if (result.startsWith(SUPPLY)) {
                    supplySum += Integer.parseInt(result.substring(SUPPLY.length() + 1));
                }
                if (result.startsWith(BUY)) {
                    buySum += Integer.parseInt(result.substring(BUY.length() + 1));
```

**Review by zachar-chorny — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,96 @@
...
+        List<String> list = new ArrayList<>();
+        while (value != null) {
+            list.add(value);
+            try {
+                value = bufferedReader.readLine();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return list;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:49]:
```suggestion
        Path path = Paths.get(fromMyFile);
        try {
            return Files.readAllLines(path);
        } catch (IOException e) {
            throw new RuntimeException("Can't read file " + fromMyFile + ".", e);
        }
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 8]
```diff
@@ -1,7 +1,96 @@
...
+                .append(SUPPLY).append(SEPARATOR)
+                .append(supplySum)
+                .append(System.lineSeparator())
+                .append(BUY).append(SEPARATOR).append(buySum)
+                .append(System.lineSeparator()).append(RESULT).append(SEPARATOR)
+                .append(resultSum).toString();
+    }
+
+    private void writeToFile(String report, String toFileName) {
+        File toMyFile = new File(toFileName);
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:79]: You can cut down a lot here too, search for how)

**Review by zachar-chorny — CHANGES_REQUESTED** [review]

**Review by aleksandr-bevza — APPROVED** [review]

**Pull request closed without merging**